src/vos/vos_aggregate.c

```diff
@@ -1345,8 +1345,14 @@ unmark_removals(struct agg_merge_window *mw, const struct agg_phy_ent *phy_ent)
 		if (rmv_ent->re_rect.rc_ex.ex_lo > phy_ent->pe_rect.rc_ex.ex_hi)
 			continue;
 
-		D_ASSERT(rmv_ent->re_phy_count > 0);
-		rmv_ent->re_phy_count--;
+		/*
+		 * Aggregation could fail out before processing the invisible record
+		 * which being covered by a removal record (see join_merge_window()),
+		 * in such case, removal record & physical record are both enqueued
+		 * but the removal record isn't referenced yet.
+		 */
+		if (rmv_ent->re_phy_count > 0)
```

Comment:
may D_ASSERT(rmv_ent->re_phy_count >= 0) ? as the count is int

Reply:
yeah, it could be changed to "unsigned int".

Reply:
I was trying and failing to create any scenario in a unit test to reproduce this.  See https://daosio.atlassian.net/jira/software/c/projects/DAOS/issues/DAOS-14051 which is a duplicate of the issue, I think.

Reply:
oh, I wasn't aware that it's a known issue and you've looked into it before, then I'm wondering if my fix is incorrect. :) Is the logical record covered by a removal record must be processed immediately after the removal record? If so, I suppose the removal record must always be referenced since aggregation will never fail in the middle.

Reply:
The log in DAOS-14051 has more useful information, the log showed aggregation aborted immediately after processing the removal record (it happened to exhaust credits after processing the removal record, and the yield function agg_rate_ctl() returns -1 to inform aggregation to abort, I guess rebuild was triggered on some node of cluster, and with Di's recent patch, aggregation on all nodes has to be aborted due to the rebuild).

Let me update the PR with proper fix.